### ISSUE TITLE
About PolyScheduler in partial fc of arcface

### ISSUE THREAD

**Issue by Icecream-blue-sky** [deepinsight/insightface#2085]
I found that partial_fc use the polyScheduler to adjust the learning rate. And the learning rate is changed by epoch(not iterations):
```python
def get_lr(self):
        if self.last_epoch == -1:
            return [self.warmup_lr_init for _ in self.optimizer.param_groups]
        if self.last_epoch < self.warmup_steps:
            return self.get_warmup_lr()
        else:
            alpha = pow(
                1
                - float(self.last_epoch - self.warmup_steps)
                / float(self.max_steps - self.warmup_steps),
                self.power,
            )
            return [self.base_lr * alpha for _ in self.optimizer.param_groups]
```
So the learning rate should be changed with epoch. Am I right? However, I find that the learning rate is actually changed with iterations, why ?
![image](https://user-images.githubusercontent.com/26771037/187022311-9aee7cdc-3906-4756-9687-baf8d139769e.png)

**Comment by nttstar** [comment]
The 'last_epoch' variable does not always mean 'epoch', it depends on how often you call the method 'scheduler.step()'

**Comment by Icecream-blue-sky** [comment]
> The 'last_epoch' variable does not always mean 'epoch', it depends on how often you call the method 'scheduler.step()'

Got it.
